src/Symfony/Bridge/Doctrine/Security/User/UserLoaderInterface.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Security\User;
+
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
+
+/**
+ * Represents a class that loads UserInterface objects from Doctrine source for the authentication system.
+ *
+ * This interface is meant to facilitate the loading of a User from Doctrine source using a custom method.
+ * If you want to implement your own logic of retrieving the user from Doctrine your repository should implement this
+ * interface.
+ *
+ * @see UserInterface
+ *
+ * @author Michal Trojanowski <michal@kmt-studio.pl>
+ */
+interface UserLoaderInterface
+{
+    /**
+     * Loads the user for the given username.
+     *
+     * This method must throw UsernameNotFoundException if the user is not
+     * found.
+     *
+     * @param string $username The username
+     *
+     * @return UserInterface
+     *
+     * @see UsernameNotFoundException
+     *
+     * @throws UsernameNotFoundException if the user is not found
```

Comment:
I'm wondering whether we should ask people to throw this exception themselves, or just return ``UserInterface|null``. the EntityUserProvider already handles the case of ``null`` anyway, and this would be more in line with the way Doctrine return values work (this would allow to implement it as a ``findOneBy()`` call for instance)